superset/common/query_context.py

```diff
@@ -46,6 +47,7 @@ class QueryContext:
     enforce_numerical_metrics: ClassVar[bool] = True
 
     datasource: BaseDatasource
+    slice_id: Optional[int] = None
```

Comment:
Not sure if I'm missing something, but I don't see this being referenced anywhere. Shouldn't we be fine with just having `form_data` as that contains `slice_id`?

Reply:
..I'm assuming this should be `slice: Optional[Slice]`?
```suggestion
    slice: Optional[Slice] = None
```